File: cpp-tiny/lib/Models/AddSubscriptionsResponseAllOf.cpp


#include "AddSubscriptionsResponse_allOf.h"

using namespace Tiny;

AddSubscriptionsResponse_allOf::AddSubscriptionsResponse_allOf()
{
	result = null;
	msg = null;
	subscribed = null<std::list>();
	already_subscribed = null<std::list>();
	unauthorized = std::list<std::string>();
}

AddSubscriptionsResponse_allOf::AddSubscriptionsResponse_allOf(std::string jsonString)
{
	this->fromJson(jsonString);
}

AddSubscriptionsResponse_allOf::~AddSubscriptionsResponse_allOf()
{

}

void
AddSubscriptionsResponse_allOf::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *resultKey = "result";
    
    if(object.has_key(resultKey)) 
    {
        bourne::json value = object[resultKey];


        

        AnyType* obj = &result;
		obj->fromJson(value.dump());

    }

    const char *msgKey = "msg";
    
    if(object.has_key(msgKey)) 
    {
        bourne::json value = object[msgKey];


        

        AnyType* obj = &msg;
		obj->fromJson(value.dump());

    }

    const char *subscribedKey = "subscribed";
    
    if(object.has_key(subscribedKey)) 
    {
        bourne::json value = object[subscribedKey];


    }

    const char *already_subscribedKey = "already_subscribed";
    
    if(object.has_key(already_subscribedKey)) 
    {
        bourne::json value = object[already_subscribedKey];


    }

    const char *unauthorizedKey = "unauthorized";
    
    if(object.has_key(unauthorizedKey)) 
    {
        bourne::json value = object[unauthorizedKey];


        std::list<std::string> unauthorized_list;
        std::string element;
        for(auto& var : value.array_range())
        {

            jsonToValue(&element, var, "std::string");
            
            
            unauthorized_list.push_back(element);
        }
        unauthorized = unauthorized_list;


    }


}

bourne::json
AddSubscriptionsResponse_allOf::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["result"] = getResult().toJson();

    




	object["msg"] = getMsg().toJson();

    


    


    
    
    
    std::list<std::string> unauthorized_list = getUnauthorized();
    bourne::json unauthorized_arr = bourne::json::array();

    for(auto& var : unauthorized_list)
    {
        unauthorized_arr.append(var);
    }
    object["unauthorized"] = unauthorized_arr;

    




    return object;

}

AnyType
AddSubscriptionsResponse_allOf::getResult()
{
	return result;
}

void
AddSubscriptionsResponse_allOf::setResult(AnyType  result)
{
	this->result = result;
}

AnyType
AddSubscriptionsResponse_allOf::getMsg()
{
	return msg;
}

void
AddSubscriptionsResponse_allOf::setMsg(AnyType  msg)
{
	this->msg = msg;
}

Map<string, string>
AddSubscriptionsResponse_allOf::getSubscribed()
{
	return subscribed;
}

void
AddSubscriptionsResponse_allOf::setSubscribed(Map <string, string> subscribed)
{
	this->subscribed = subscribed;
}

Map<string, string>
AddSubscriptionsResponse_allOf::getAlreadySubscribed()
{
	return already_subscribed;
}

void
AddSubscriptionsResponse_allOf::setAlreadySubscribed(Map <string, string> already_subscribed)
{
	this->already_subscribed = already_subscribed;
}

std::list<std::string>
AddSubscriptionsResponse_allOf::getUnauthorized()
{
	return unauthorized;
}

void
AddSubscriptionsResponse_allOf::setUnauthorized(std::list <std::string> unauthorized)
{
	this->unauthorized = unauthorized;
}

File: spring/src/main/java/org/openapitools/model/AddSubscriptionsResponseAllOf.java
package org.openapitools.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * AddSubscriptionsResponseAllOf
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2021-08-10T11:47:39.922565Z[Etc/UTC]")
public class AddSubscriptionsResponseAllOf   {
  @JsonProperty("result")
  private JsonNullable<Object> result = JsonNullable.undefined();

  @JsonProperty("msg")
  private JsonNullable<Object> msg = JsonNullable.undefined();

  @JsonProperty("subscribed")
  @Valid
  private Map<String, List<String>> subscribed = null;

  @JsonProperty("already_subscribed")
  @Valid
  private Map<String, List<String>> alreadySubscribed = null;

  @JsonProperty("unauthorized")
  @Valid
  private List<String> unauthorized = null;

  public AddSubscriptionsResponseAllOf result(Object result) {
    this.result = JsonNullable.of(result);
    return this;
  }

  /**
   * Get result
   * @return result
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getResult() {
    return result;
  }

  public void setResult(JsonNullable<Object> result) {
    this.result = result;
  }

  public AddSubscriptionsResponseAllOf msg(Object msg) {
    this.msg = JsonNullable.of(msg);
    return this;
  }

  /**
   * Get msg
   * @return msg
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getMsg() {
    return msg;
  }

  public void setMsg(JsonNullable<Object> msg) {
    this.msg = msg;
  }

  public AddSubscriptionsResponseAllOf subscribed(Map<String, List<String>> subscribed) {
    this.subscribed = subscribed;
    return this;
  }

  public AddSubscriptionsResponseAllOf putSubscribedItem(String key, List<String> subscribedItem) {
    if (this.subscribed == null) {
      this.subscribed = new HashMap<>();
    }
    this.subscribed.put(key, subscribedItem);
    return this;
  }

  /**
   * A dictionary where the key is the email address of the user/bot and the value is a list of the names of the streams that were subscribed to as a result of the query. 
   * @return subscribed
  */
  @ApiModelProperty(value = "A dictionary where the key is the email address of the user/bot and the value is a list of the names of the streams that were subscribed to as a result of the query. ")

  @Valid

  public Map<String, List<String>> getSubscribed() {
    return subscribed;
  }

  public void setSubscribed(Map<String, List<String>> subscribed) {
    this.subscribed = subscribed;
  }

  public AddSubscriptionsResponseAllOf alreadySubscribed(Map<String, List<String>> alreadySubscribed) {
    this.alreadySubscribed = alreadySubscribed;
    return this;
  }

  public AddSubscriptionsResponseAllOf putAlreadySubscribedItem(String key, List<String> alreadySubscribedItem) {
    if (this.alreadySubscribed == null) {
      this.alreadySubscribed = new HashMap<>();
    }
    this.alreadySubscribed.put(key, alreadySubscribedItem);
    return this;
  }

  /**
   * A dictionary where the key is the email address of the user/bot and the value is a list of the names of the streams that the user/bot is already subscribed to. 
   * @return alreadySubscribed
  */
  @ApiModelProperty(value = "A dictionary where the key is the email address of the user/bot and the value is a list of the names of the streams that the user/bot is already subscribed to. ")

  @Valid

  public Map<String, List<String>> getAlreadySubscribed() {
    return alreadySubscribed;
  }

  public void setAlreadySubscribed(Map<String, List<String>> alreadySubscribed) {
    this.alreadySubscribed = alreadySubscribed;
  }

  public AddSubscriptionsResponseAllOf unauthorized(List<String> unauthorized) {
    this.unauthorized = unauthorized;
    return this;
  }

  public AddSubscriptionsResponseAllOf addUnauthorizedItem(String unauthorizedItem) {
    if (this.unauthorized == null) {
      this.unauthorized = new ArrayList<>();
    }
    this.unauthorized.add(unauthorizedItem);
    return this;
  }

  /**
   * A list of names of streams that the requesting user/bot was not authorized to subscribe to.  Only present if `authorization_errors_fatal=false`. 
   * @return unauthorized
  */
  @ApiModelProperty(value = "A list of names of streams that the requesting user/bot was not authorized to subscribe to.  Only present if `authorization_errors_fatal=false`. ")


  public List<String> getUnauthorized() {
    return unauthorized;
  }

  public void setUnauthorized(List<String> unauthorized) {
    this.unauthorized = unauthorized;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    AddSubscriptionsResponseAllOf addSubscriptionsResponseAllOf = (AddSubscriptionsResponseAllOf) o;
    return Objects.equals(this.result, addSubscriptionsResponseAllOf.result) &&
        Objects.equals(this.msg, addSubscriptionsResponseAllOf.msg) &&
        Objects.equals(this.subscribed, addSubscriptionsResponseAllOf.subscribed) &&
        Objects.equals(this.alreadySubscribed, addSubscriptionsResponseAllOf.alreadySubscribed) &&
        Objects.equals(this.unauthorized, addSubscriptionsResponseAllOf.unauthorized);
  }

  @Override
  public int hashCode() {
    return Objects.hash(result, msg, subscribed, alreadySubscribed, unauthorized);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class AddSubscriptionsResponseAllOf {\n");
    
    sb.append("    result: ").append(toIndentedString(result)).append("\n");
    sb.append("    msg: ").append(toIndentedString(msg)).append("\n");
    sb.append("    subscribed: ").append(toIndentedString(subscribed)).append("\n");
    sb.append("    alreadySubscribed: ").append(toIndentedString(alreadySubscribed)).append("\n");
    sb.append("    unauthorized: ").append(toIndentedString(unauthorized)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

